test/semantics/tensor.py

```diff
@@ -260,6 +260,13 @@ def test_Tensor_adjoint_eval():
     assert tensor1 == tensor2
 
 
+def test_Tensor_dtype_inference():
+    assert Box("F(A)", Dim(1), Dim(1), data=None) == Box("F(A)", Dim(1), Dim(1))
+    assert Box("X", Dim(1), Dim(1), data=[1]) == Box[np.int64]("X", Dim(1), Dim(1), data=[1])
```

Comment:
Easy to change the test to check that the problem above doesn't happen

```suggestion
    assert Box("X", Dim(1), Dim(1), data=[0]) == Box[np.int64]("X", Dim(1), Dim(1), data=[0])
```